zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -1176,27 +1179,27 @@ impl<F: Field> ExecutionConfig<F> {
                         break;
                     }
                     second_last_real_step = Some(cur);
+                    // record offset of current step before assignment
+                    second_last_real_step_offset = offset;
                     offset = assign_padding_or_step(cur, offset, next.copied(), None)?;
                 }
-                let second_last_real_step_offset = offset;
 
                 // next step priority: padding > end_chunk > end_block
                 let mut next_step_after_real_step = None;
 
                 // part2: assign padding
-                if let Some(padding) = &chunk.padding {
```

Comment:
Actually padding is a must, so it will be best to remove `Option` type.
But we still keep chunk.padding as `Option` due to there are 2 plae to set this variable in `bus-mapping`. 1 is `end_chunk`, the others is `end_block`